packages/covenant-utils/src/lib.rs

```diff
@@ -297,22 +310,34 @@ impl CovenantPartiesConfig {
     pub fn get_response_attributes(self) -> Vec<Attribute> {
         let mut attrs = vec![
             Attribute::new("party_a_address", self.party_a.addr),
-            Attribute::new("party_a_denom", self.party_a.provided_denom),
+            Attribute::new("party_a_ibc_denom", self.party_a.ibc_denom),
             Attribute::new("party_b_address", self.party_b.addr),
-            Attribute::new("party_b_denom", self.party_b.provided_denom),
+            Attribute::new("party_b_ibc_denom", self.party_b.ibc_denom),
         ];
         attrs.extend(
             self.party_a
-                .refund_config
+                .receiver_config
                 .get_response_attributes("party_a_".to_string()),
         );
         attrs.extend(
             self.party_b
-                .refund_config
+                .receiver_config
                 .get_response_attributes("party_b_".to_string()),
         );
         attrs
     }
+
+    pub fn match_caller_party(&self, caller: String) -> Result<CovenantParty, StdError> {
+        let a = self.clone().party_a;
+        let b = self.clone().party_b;
+        if a.addr == caller {
+            Ok(a)
+        } else if b.addr == caller {
+            Ok(b)
+        } else {
+            Err(StdError::generic_err("unauthorized"))
+        }
```

Comment:
You should be able to pass `&caller` to the function, and then avoid the clone and make it much simpler.